include/fmt/base.h

```diff
@@ -1084,6 +1101,7 @@ template <typename T, typename Char,
           FMT_ENABLE_IF(is_static_named_arg<T>::value)>
 FMT_CONSTEXPR void init_static_named_arg(named_arg_info<Char>* named_args,
                                          int& arg_index, int& named_arg_index) {
+  check_for_duplicate(named_args, named_arg_index, T::name);
```

Comment:
I wonder if calling `check_for_duplicate` only if `named_arg_index` is nonzero fixes the uninitialized warning.